tests/s3/versioning/test_get_put_object_tagging.py

```diff
@@ -0,0 +1,273 @@
+        self.s3_tag_obj = S3TaggingTestLib(endpoint_url=S3_CFG["s3_url"])
+        self.test_dir_path = os.path.join(TEST_DATA_FOLDER, "TestGetPutObjectTagging")
+        if not path_exists(self.test_dir_path):
+            make_dirs(self.test_dir_path)
+            LOGGER.info("Created path: %s", self.test_dir_path)
+        self.file_path = os.path.join(self.test_dir_path, f"del_obj_tag_{time.perf_counter_ns()}")
+        create_file(fpath=self.file_path, count=1)
+        LOGGER.info("Created file: %s", self.file_path)
+        self.bucket_name = f"tag-bkt-{time.perf_counter_ns()}"
+        self.object_name = f"tag-obj-{time.perf_counter_ns()}"
+        self.versions = dict()
+        self.ver_tag = dict()
+        self.ver_tag.update({self.object_name: dict()})
+        res = self.s3_test_obj.create_bucket(self.bucket_name)
+        assert_utils.assert_true(res[0], res[1])
+        assert_utils.assert_equal(res[1], self.bucket_name, res[1])
+        LOGGER.info("Created a bucket with name : %s", self.bucket_name)
+
+    def teardown_method(self):
+        """
+        Function will be performed cleanup after each test case.
+        """
+        LOGGER.info("STARTED: Teardown operations")
+        LOGGER.info("Clean : %s", self.test_dir_path)
+        if path_exists(self.file_path):
+            res = remove_file(self.file_path)
+            LOGGER.info("cleaned path: %s, res: %s", self.file_path, res)
+        if path_exists(self.test_dir_path):
+            remove_dirs(self.test_dir_path)
+        LOGGER.info("Cleanup test directory: %s", self.test_dir_path)
+        # DELETE Object with VersionId is WIP, uncomment once feature is available
+        # res = self.s3_test_obj.bucket_list()
+        # pref_list = []
+        # for bucket_name in res[1]:
+        #     if bucket_name.startswith("tag-bkt"):
+        #         empty_versioned_bucket(self.s3_ver_obj, bucket_name)
+        #         pref_list.append(bucket_name)
+        # if pref_list:
+        #     res = self.s3_test_obj.delete_multiple_buckets(pref_list)
+        #     assert_utils.assert_true(res[0], res[1])
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-40429")
+    @CTFailOn(error_handler)
+    def test_get_put_obj_tags_40429(self):
+        """Test GET and PUT object tagging for pre-existing object in a versioning enabled
+        bucket"""
+
+        LOGGER.info("STARTED: Test GET and PUT object tagging for pre-existing object in"
+                    " a versioning enabled bucket ")
+
+        LOGGER.info("Step 1: Upload object %s before enabling versioning on bucket %s",
+                    self.object_name, self.bucket_name)
+        upload_version(self.s3_test_obj, bucket_name=self.bucket_name, file_path=self.file_path,
+                       object_name=self.object_name, versions_dict=self.versions,
+                       is_unversioned=True)
+        latest_v = self.versions[self.object_name]["version_history"][-1]
```

Comment:
maybe latest_v needs to be taken after bucket versioning is enabled?
cc: @apoorva-rao